### ISSUE TITLE
autoreconf fails on all packages since sbang rework

### ISSUE THREAD

**Issue by chuckatkins** [LLNL/spack#19580]
Ever since #19529 the `autoreconf` step is failing on every package I've tried it on.  It seems as though some quoting is happening that's failing to separate arguments.  The example here is for numactl but I get a similar failure in autoreconf for any package that needs it.

```
$ spack install -v -j56 numactl
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libsigsegv-2.12-q6vrhvmsoctafa6kk5nlaun73zoo52dj
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/berkeley-db-18.1.40-4j2gl4a5obnc6dpdhrdeq7rlj5ozmt6b
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/pkgconf-1.7.3-fdbjumxg4ahzsyv2ssrnfojz3632bnb2
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/m4-1.4.18-5kyc5qjbbztwlsyf254d377baxd2sq4e
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/ncurses-6.2-leymoflebjh5syaxoqjsyxym4ikvwtta
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libtool-2.4.6-l7e4jkyv5kirijfp3b2owlqnkx5lx5ma
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/readline-8.0-b6nc7v2wkwc6mpng6b5runxosjzx6y4v
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/gdbm-1.18.1-aiqqwmgsnyg6gsocoff2hkers3rdh4jq
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/perl-5.30.3-i5otw34rjf6zwl2eobajtelpeofyusgz
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/autoconf-2.69-dizixbfyqb7xxrvbv3stkg3kfwbsppgv
[+] /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/automake-1.16.2-ukuwmhh36zcs6yngsgzftxlpb3gq5gi4
==> Installing numactl
==> No binary for numactl found: installing from source
==> Using cached archive: /home/genericuser/Code/spack/var/spack/cache/_source-cache/archive/1e/1ee27abd07ff6ba140aaf9bc6379b37825e54496e01d6f7343330cf1a4487035.tar.gz
==> numactl: Executing phase: 'autoreconf'
==> [2020-10-28-16:39:01.197734] Configure script not found: trying to generate it
==> [2020-10-28-16:39:01.197924] Warning: *********************************************************
==> [2020-10-28-16:39:01.198016] Warning: * If the default procedure fails, consider implementing *
==> [2020-10-28-16:39:01.198080] Warning: *        a custom AUTORECONF phase in the package       *
==> [2020-10-28-16:39:01.198137] Warning: *********************************************************
==> [2020-10-28-16:39:01.199510] '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/autoconf-2.69-dizixbfyqb7xxrvbv3stkg3kfwbsppgv/bin/autoreconf' '-ivf' '-I' '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/automake-1.16.2-ukuwmhh36zcs6yngsgzftxlpb3gq5gi4/share/aclocal' '-I' '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libtool-2.4.6-l7e4jkyv5kirijfp3b2owlqnkx5lx5ma/share/aclocal' 
Can't open perl script "/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/autoconf-2.69-dizixbfyqb7xxrvbv3stkg3kfwbsppgv/bin/autoreconf -ivf -I /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/automake-1.16.2-ukuwmhh36zcs6yngsgzftxlpb3gq5gi4/share/aclocal -I /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libtool-2.4.6-l7e4jkyv5kirijfp3b2owlqnkx5lx5ma/share/aclocal": No such file or directory
==> Error: ProcessError: Command exited with status 2:
    '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/autoconf-2.69-dizixbfyqb7xxrvbv3stkg3kfwbsppgv/bin/autoreconf' '-ivf' '-I' '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/automake-1.16.2-ukuwmhh36zcs6yngsgzftxlpb3gq5gi4/share/aclocal' '-I' '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libtool-2.4.6-l7e4jkyv5kirijfp3b2owlqnkx5lx5ma/share/aclocal'
     
1 error found in build log:
     2    ==> [2020-10-28-16:39:01.197734] Configure script not found: trying to generate it
     3    ==> [2020-10-28-16:39:01.197924] Warning: *********************************************************
     4    ==> [2020-10-28-16:39:01.198016] Warning: * If the default procedure fails, consider implementing *
     5    ==> [2020-10-28-16:39:01.198080] Warning: *        a custom AUTORECONF phase in the package       *
     6    ==> [2020-10-28-16:39:01.198137] Warning: *********************************************************
     7    ==> [2020-10-28-16:39:01.199510] '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/autoconf-2.69-dizixbfyqb7xxrvbv3stkg3kfwbsppgv/bin/autoreconf' '-ivf' '-I' '/home/genericuser/Code/spack/opt/spack/linux-centos7-has
          well/gcc-8.3.0/automake-1.16.2-ukuwmhh36zcs6yngsgzftxlpb3gq5gi4/share/aclocal' '-I' '/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libtool-2.4.6-l7e4jkyv5kirijfp3b2owlqnkx5lx5ma/share/aclocal'
  >> 8    Can't open perl script "/home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/autoconf-2.69-dizixbfyqb7xxrvbv3stkg3kfwbsppgv/bin/autoreconf -ivf -I /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/a
          utomake-1.16.2-ukuwmhh36zcs6yngsgzftxlpb3gq5gi4/share/aclocal -I /home/genericuser/Code/spack/opt/spack/linux-centos7-haswell/gcc-8.3.0/libtool-2.4.6-l7e4jkyv5kirijfp3b2owlqnkx5lx5ma/share/aclocal": No such file or directory

See build log for details:
  /tmp/genericuser/spack-stage/spack-stage-numactl-2.0.14-kfksu6apiic5vhgnaglymyxn3ynuwhad/spack-build-out.txt

$
```
Reverting #19529 restores correct functionality

**Comment by chuckatkins** [comment]
@tgamblin ?

**Comment by chuckatkins** [comment]
Fixed by #19582
